chain/src/chain.rs

```diff
@@ -494,14 +504,20 @@ impl ChainService {
 
         if new_best_block {
             let tip_header = block.header();
-            info!(
-                "block: {}, hash: {:#x}, epoch: {:#}, total_diff: {:#x}, txs: {}",
-                tip_header.number(),
-                tip_header.hash(),
-                tip_header.epoch(),
-                total_difficulty,
-                block.transactions().len()
-            );
+
+            // if in IBD, update statistics and print the 1st block in the epoch
+            if self.shared.is_initial_block_download() {
+                self.update_print_statics(&block, &tip_header, total_difficulty.clone());
+            } else {
+                info!(
+                    "block: {}, hash: {:#x}, epoch: {:#}, total_diff: {:#x}, txs: {}",
+                    tip_header.number(),
+                    tip_header.hash(),
+                    tip_header.epoch(),
+                    total_difficulty,
+                    block.transactions().len()
+                );
+            }
```

Comment:
suggest to move all log related code to a single fn, it can reduce duplicate code, for example  `info!("block: {}...`

Reply:
i just move related info_block call into standalone functions, but not all functions can be merged into single function, due to message format.